packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -689,19 +686,25 @@ export class Cluster extends Resource implements ICluster {
       new CfnOutput(this, 'ClusterName', { value: this.clusterName });
     }
 
-    // map the IAM role to the `system:masters` group.
-    if (props.mastersRole) {
-      if (!this.kubectlEnabled) {
+    if (!this.kubectlEnabled) {
+      if (props.mastersRole) {
         throw new Error('Cannot specify a "masters" role if kubectl is disabled');
       }
+    } else {
+      // if an explicit role is not configured, define a masters role that can
+      // be assumed by anyone in the account (with sts:AssumeRole permissions of
```

Comment:
I would mention the `sts:AssumeRole` explanation in the README, and possibly even in the default value for `mastersRole`. It was a bit vague for me until I reached this point.

Reply:
Done